test/unit/lookup_key_test.rb

```diff
@@ -24,7 +24,7 @@ def test_path2match_single_domain_path
     key   = ""
     value = ""
     as_admin do
-      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first)
+      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first, :key_type => "string")
```

Comment:
I began this PR with the default behaviour in caster.rb:46 being that it returns the value. you requested this to be changed. now you're requesting this to be switched back.

I'm confused now. What is the expected behaviour? why did you want me to change it? why do you want me to change it back now?